src/Aspire.Dashboard/Otlp/Http/OtlpHttpEndpointsBuilder.cs

```diff
@@ -0,0 +1,205 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Net.Http.Headers;
+using System.Reflection;
+using Aspire.Dashboard.Authentication;
+using Google.Protobuf;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp.Http;
+
+public static class OtlpHttpEndpointsBuilder
+{
+    public const string ProtobufContentType = "application/x-protobuf";
+    public const string JsonContentType = "application/json";
+
+    public static void MapHttpOtlpApi(this IEndpointRouteBuilder endpoints)
+    {
+        var group = endpoints
+            .MapGroup("/v1")
+            .AddOtlpHttpMetadata();
+
+        group.MapPost("logs", static (MessageBindable<ExportLogsServiceRequest> request, OtlpLogsService service) =>
+        {
+            if (request.Message == null)
+            {
+                return Results.Empty;
+            }
+            return OtlpResult.Response(service.Export(request.Message));
+        });
+        group.MapPost("traces", static (MessageBindable<ExportTraceServiceRequest> request, OtlpTraceService service) =>
+        {
+            if (request.Message == null)
+            {
+                return Results.Empty;
+            }
+            return OtlpResult.Response(service.Export(request.Message));
+        });
+        group.MapPost("metrics", (MessageBindable<ExportMetricsServiceRequest> request, OtlpMetricsService service) =>
+        {
+            if (request.Message == null)
+            {
+                return Results.Empty;
+            }
+            return OtlpResult.Response(service.Export(request.Message));
+        });
+    }
+
+    private enum KnownContentType
+    {
+        None,
+        Protobuf,
+        Json
+    }
+
+    private static KnownContentType GetKnownContentType(string? contentType, out StringSegment charSet)
+    {
+        if (contentType != null && MediaTypeHeaderValue.TryParse(contentType, out var mt))
+        {
+            if (string.Equals(mt.MediaType, JsonContentType, StringComparison.OrdinalIgnoreCase))
```

Comment:
Could introduce `StringComparisons.MediaType` for this and the comparison below.

Reply:
I think this is a one off usage.